samples/financial_transcripts/generate_benchmark_qa.ipynb

```diff
@@ -0,0 +1,562 @@
+    "        {chunk_text1}\n",
+    "        input_text2:\n",
+    "        {chunk_text2}\n",
+    "        ticker: {ticker}\n",
+    "        quarter: {quarter}\n",
+    "        year: {year}\n",
+    "        \"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Randomly pick filter parameters:\n",
+    "\n",
+    "Add Filter Parameters to randomly pick them for extracting context (chunks). This will help diversify generating questions and answers. \n",
+    "In the MSFT Financial Transcripts use-case, the Ticker name is MSFT, but you can easily add other ticker labels for a larger Financial dataset. `Year`,`Quarter`, and `Id` are the key parameters used in this use-case. In this notebook, we are using two random chunks (chunk ids) from specified filter keys (year, quarter).\n",
+    "\n",
+    "###### TODO: Add Tools (an updated version of function calls) once it is available for working with newer models. Function Calls are currently deprecated for gpt models with versions beyond 07-01-2023."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Ticker = np.random.choice(df['Ticker'].unique())\n",
+    "Year = np.random.choice(df[df['Ticker']==Ticker]['Year'].unique())\n",
+    "Quarter = np.random.choice(df[(df['Ticker']==Ticker) & (df['Year']==Year)]['Quarter'].unique())\n",
+    "Id = np.random.choice(df[(df['Quarter']==Quarter) & (df['Year']==Year)]['Id'].unique())\n",
+    "Id2 = np.random.choice(df[(df['Quarter']==Quarter) & (df['Year']==Year) & (df['Id']!=Id)]['Id'].unique())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'MSFT'"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "Ticker"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Generate Questions (using Azure OpenAI only)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1. What was the percentage increase in server products and cloud services revenue for MSFT in Q1 FY23?\n",
+      "Answer: The server products and cloud services revenue increased by 22% in constant currency for MSFT in Q1 FY23.\n",
+      "\n",
+      "2. How much did Azure and other cloud services revenue grow in constant currency for MSFT in Q1 FY23?\n",
+      "Answer: Azure and other cloud services revenue grew 35% in constant currency for MSFT in Q1 FY23.\n",
+      "\n",
+      "3. Did Azure consumption growth meet expectations for MSFT in Q1 FY23?\n",
+      "Answer: No, Azure consumption growth was about a point lower than expected for MSFT in Q1 FY23.\n",
+      "\n",
```

Comment:
This may be not completely correct, because the question is asked about Azure only, and the estimate of being about a point lower than expected is given for Azure and other cloud services:

"Azure and other cloud services revenue grew 35 percent and 42 percent in constant currency, about a point lower than expected"

Reply:
Yep, I am not expecting generation to be correct. There are some known reasons for that. The chunking size is too little. If you look at number of tokens/characters, for example. This is why I referenced the like to the other notebook in promptflow-sample: https://github.com/microsoft/promptflow-rag-samples/blob/main/financial_transcripts/preprocessing/generate_benchmark_qa.ipynb. That seems to improve if larger chunks are used (page as context, which is larger). Are there other possible reasons you could think of?